commands/run.sh

```diff
@@ -275,10 +275,14 @@ if [[ -n "${BUILDKITE_AGENT_ACCESS_TOKEN:-}" ]] ; then
   if [[ "$(plugin_read_config CHECK_LINKED_CONTAINERS "true")" != "false" ]] ; then
 
     # Get list of failed containers
-    failed_containers=($(
-      docker inspect -f '{{if ne 0 .State.ExitCode}}{{.Name}}.{{.State.ExitCode}}{{ end }}' \
-      $(docker_ps_by_project -q)
-    ))
+    containers=($(docker_ps_by_project -q))
+    failed_containers=()
+    if [[ 0 != "${#containers[@]}" ]] ; then
+      failed_containers=($(
```

Comment:
This is problematic, it's relying on implicit token splitting from the output of "$(.." into an array. 

Check out https://github.com/koalaman/shellcheck/wiki/SC2207, would suggest something like:

```
failed_containers=()
while IFS='' read -r line; do 
  array+=("$line") 
done < <(docker inspect -f '{{if ne 0 .State.ExitCode}}{{.Name}}.{{.State.ExitCode}}{{ end }}' "${containers[@]}")
```

Reply:
@lox as mentioned on #210 that breaks BATS testing, that is why I'm using the other method.